server/src/main/java/org/elasticsearch/action/admin/indices/resolve/TransportResolveClusterAction.java

```diff
@@ -0,0 +1,260 @@
+
+public class TransportResolveClusterAction extends HandledTransportAction<ResolveClusterActionRequest, ResolveClusterActionResponse> {
+
+    private static final Logger logger = LogManager.getLogger(TransportResolveClusterAction.class);
+
+    public static final String NAME = "indices:admin/resolve/cluster";
+    public static final ActionType<ResolveClusterActionResponse> TYPE = new ActionType<>(NAME, ResolveClusterActionResponse::new);
+    public static final String ACTION_NODE_NAME = NAME + "[n]"; // TODO: do we need this?
+
+    private final ThreadPool threadPool;
+    private final Executor searchCoordinationExecutor;
+    private final ClusterService clusterService;
+    private final RemoteClusterService remoteClusterService;
+    private final IndexNameExpressionResolver indexNameExpressionResolver;
+    private final boolean ccsCheckCompatibility;
+
+    @Inject
+    public TransportResolveClusterAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(NAME, transportService, actionFilters, ResolveClusterActionRequest::new, EsExecutors.DIRECT_EXECUTOR_SERVICE);
+        this.threadPool = threadPool;
+        this.searchCoordinationExecutor = threadPool.executor(ThreadPool.Names.SEARCH_COORDINATION);
+        this.clusterService = clusterService;
+        this.remoteClusterService = transportService.getRemoteClusterService();
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+        this.ccsCheckCompatibility = SearchService.CCS_VERSION_CHECK_SETTING.get(clusterService.getSettings());
+    }
+
+    @Override
+    protected void doExecute(Task task, ResolveClusterActionRequest request, ActionListener<ResolveClusterActionResponse> listener) {
+        // workaround for https://github.com/elastic/elasticsearch/issues/97916 - TODO remove this when we can
+        searchCoordinationExecutor.execute(ActionRunnable.wrap(listener, l -> doExecuteForked(task, request, l)));
+    }
+
+    protected void doExecuteForked(Task task, ResolveClusterActionRequest request, ActionListener<ResolveClusterActionResponse> listener) {
+        if (ccsCheckCompatibility) {
+            checkCCSVersionCompatibility(request);
+        }
+        assert task instanceof CancellableTask;
+        final CancellableTask resolveClusterTask = (CancellableTask) task;
+        ClusterState clusterState = clusterService.state();
+        Map<String, OriginalIndices> remoteClusterIndices = remoteClusterService.groupIndices(request.indicesOptions(), request.indices());
+        OriginalIndices localIndices = remoteClusterIndices.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
+
+        Map<String, ResolveClusterInfo> clusterInfoMap = new ConcurrentHashMap<>();
+        if (remoteClusterIndices.size() > 0) {
```

Comment:
Don't think you need to branch here, the `if` branch works even if `remoteClusterIndices` is empty AFAICT.

Reply:
I tested it before and just now again and I believe I do need it. 

If you do 
```
GET /_resolve/cluster/b*
```
it only executes the else branch.

If you do 
```
GET /_resolve/cluster/*,remote1:remote2:*,remote2:*
```
(and remote1 has remote2 configured as a remote), both local and remote1 enter the if block so that they can call out to the specified remote, where as remote2 only executes the else block.

Reply:
That sounds accurate (although I think we should actively forbid `remote1:remote2:*`) but still unnecessary. Both branches do the same thing if `remoteClusterIndices` is empty, so there's no need to branch.

Reply:
I see what you mean. You're right. Simplified in next push. Thanks.